DfciPkg/Library/DfciRecoveryLib/DfciRecoveryLib.c

```diff
@@ -76,8 +76,9 @@ GetRecoveryChallenge (
   // Allocate the buffer...
   if (!EFI_ERROR (Status)) {
     NewChallenge = AllocatePool (sizeof (DFCI_RECOVERY_CHALLENGE) + DFCI_MULTI_STRING_MAX_SIZE);
+    // Exit if we ran out of resources

```

Comment:
I know it is unrelated to your direct code change but I think it should be updated since you're already editing here.

This line:

```c
  DEBUG ((DEBUG_VERBOSE, "%a: LocateProtocol(RNG) = %r\n", __FUNCTION__, Status));
```

Is almost useless as is. Can you add an `else` block (to `if (!EFI_ERROR (Status)) {`) and add the message there with debug level `DEBUG_ERROR` (instead of `DEBUG_VERBOSE`)?

Reply:
I think that the invalid access could still happen if the LocateProtocol fails.
Should we return Status if LocateProtocol fails for RngDxe?

Reply:
Done.